public/app/features/plugins/sandbox/sandbox_plugin_loader_registry.ts

```diff
@@ -0,0 +1,83 @@
+import { PluginSignatureType } from '@grafana/data';
+import { config } from '@grafana/runtime';
+
+import { getPluginSettings } from '../pluginSettings';
+
+type SandboxEligibilityCheckParams = {
+  isAngular?: boolean;
+  pluginId: string;
+};
+
+type SandboxEnabledCheck = (params: SandboxEligibilityCheckParams) => Promise<boolean>;
+
+/**
+ * We allow core extensions to register their own
+ * sandbox enabled checks.
+ */
+const sandboxEnabledChecks: SandboxEnabledCheck[] = [isPluginFrontendSandboxEnabled];
+
+export function addSandboxEnabledCheck(checker: SandboxEnabledCheck) {
+  sandboxEnabledChecks.push(checker);
+}
+
+export async function shouldLoadPluginInFrontendSandbox({
+  isAngular,
+  pluginId,
+}: SandboxEligibilityCheckParams): Promise<boolean> {
+  // basic check if the plugin is eligible for the sandbox
+  if (!(await isPluginFrontendSandboxElegible({ isAngular, pluginId }))) {
+    return false;
+  }
```

Comment:
nit: why don't you set this as a `checker`?

Reply:
I thought about it and decided it to separate the concept of "eligibility" with the concept of "enabled". a plugin could be enabled by not be elegible to run in the sandbox. e.g: the user puts an angular plugin in the list of enabled plugins, this won't run.